src/utils/registration-helpers.ts

```diff
@@ -693,9 +695,34 @@ export const willResetNviStatuses = (persistedRegistration: Registration, update
   }
 
   const hasChangedCategory =
-    persistedRegistration.entityDescription?.reference?.publicationInstance.type !==
-    updatedRegistration.entityDescription?.reference?.publicationInstance.type;
+    persistedRegistration.entityDescription?.reference?.publicationContext.type !==
+    updatedRegistration.entityDescription?.reference?.publicationContext.type;
   if (hasChangedCategory) {
     return true;
   }
+
+  const persistedRegistrationWithJournal = persistedRegistration as JournalRegistration | ChapterRegistration;
+  const updatedRegistrationWithJournal = updatedRegistration as JournalRegistration | ChapterRegistration;
```

Comment:
dette blir kanskje mer riktig nå? siden denne også dekker bøker
```suggestion
  const persistedRegistrationWithContextId = persistedRegistration as JournalRegistration | ChapterRegistration;
  const updatedRegistrationWithContextId = updatedRegistration as JournalRegistration | ChapterRegistration;
```